tests/test_pickle_gem.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import absolute_import, print_function, division
+from six.moves import cPickle as pickle, range
+import gem
+import numpy
+import pytest
+
+
+@pytest.mark.parametrize('protocol', range(3))
+def test_pickle_gem(protocol):
+    f = gem.VariableIndex(gem.Indexed(gem.Variable('facet', (2,)), (1,)))
+    q = gem.Index()
+    r = gem.Index()
+    _1 = gem.Indexed(gem.Literal(numpy.random.rand(3, 6, 8)), (f, q, r))
+    _2 = gem.Indexed(gem.view(gem.Variable('w', (None, None)), slice(8), slice(1)), (r, 0))
+    expr = gem.ComponentTensor(gem.IndexSum(gem.Product(_1, _2), (r,)), (q,))
+
+    unpickled = pickle.loads(pickle.dumps(expr, protocol))
+    assert repr(expr) == repr(unpickled)
+
```

Comment:
Are you comparing `repr(expr) == repr(unpickled)` here because there's no guarantee that two objects comparing equal with `==` serialize properly?